### PR TITLE
Update test cases for ZookeeperSyncDataService

### PR DESCRIPTION
**Pull request by zendwang**
Make sure that:

- [ ] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by flwrchild42** [comment]
Did I do something I shouldn’t have again 


Debbie Slater 

On Thursday, December 3, 2020, 11:31 PM, zendwang <notifications@github.com> wrote:


Make sure that:
   
   -  You have read the contribution guidelines.
   -  You submit test cases (unit or integration tests) that back your changes.
   -  Your local test passed mvn clean install -Dmaven.javadoc.skip=true.

You can view, comment on, or merge this pull request online at:

  https://github.com/dromara/soul/pull/516

Commit Summary
   
   - Update test cases for HttpParamConverter

File Changes
   
   -  A soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java (290) 

Patch Links:
   
   - https://github.com/dromara/soul/pull/516.patch
   - https://github.com/dromara/soul/pull/516.diff

**Comment by songyuequan** [comment]
Please fix the checkStyle violations first. you can test locally by command `mvn clean install -Dmaven.javadoc.skip=true`

File: soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    public static void tearDown() throws Exception {
+        zkServer.stop();
+    }
+    
+    @SneakyThrows
+    @Test
+    public void testWatcherPlugin() {
+        PluginData pluginData = new PluginData("6", PluginEnum.DIVIDE.getName(), "aaaaa", 0, Boolean.TRUE);
+        writePlugin(pluginData);
+        final String pluginPath = ZkPathConstants.buildPluginPath(pluginData.getName());
```
Comment (dengliming) [inline soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java:103]: Can you also test `onSubscribe ` method? I think we should verify the data from watch. similar in other places. Thanks.

**Review by dengliming — CHANGES_REQUESTED** [review]

File: soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java (hunk 1 above)
Comment (zendwang) [reply soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java:103]: ok,I will do it

File: soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+            @Override
+            public void onSubscribe(final AppAuthData appAuthData) {
+                log.info("AuthDataSubscriber.onSubscribe,AppAuthData:{}", appAuthData);
+            }
+
+            @Override
+            public void unSubscribe(final AppAuthData appAuthData) {
+                log.info("AuthDataSubscriber.unSubscribe,AppAuthData:{}", appAuthData);
+            }
+        };
```
Comment (dengliming) [inline soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java:111]: Hey~ I suggest you can do like. Not just logging.
```
@SneakyThrows
    @Test
    public void testWatcherPlugin() {
        final CountDownLatch latch = new CountDownLatch(1);
        PluginData pluginData = new PluginData("6", PluginEnum.DIVIDE.getName(), "aaaaa", 0, Boolean.TRUE);
        writePlugin(pluginData);
        final List<PluginData> actuals = new ArrayList<>();
        syncDataService = new ZookeeperSyncDataService(zkClient, new PluginDataSubscriber() {
            @Override
            public void onSubscribe(PluginData pluginData) {
                latch.countDown();
                actuals.add(pluginData);
            }
        }, Collections.emptyList(), Collections.emptyList());
        latch.await(3, TimeUnit.SECONDS);
        Assert.assertEquals(1, actuals.size());
        Assert.assertEquals(pluginData, actuals.get(0));
    }
```

Comment (zendwang) [reply soul-sync-data-center/soul-sync-data-zookeeper/src/test/java/org/dromara/soul/sync/data/zookeeper/ZookeeperSyncDataServiceTest.java:111]: I will try to refactor  in this way!

**Review by dengliming — APPROVED** [review]

**Pull request merged**